tileview/src/main/java/com/qozix/tileview/TileView.java

```diff
@@ -850,4 +852,81 @@ public void submit() {
     }
   }
 
+  /**
+   * Object used to keep some data when a configuration change happens and the activity is
+   * re-created.
+   * It's boiler-plate but this is how to save View state.
+   */
+  private static class SavedState extends BaseSavedState {
+    /* This will store the current scale and position */
+    float scale;
+    int savedCenterX;
+    int savedCenterY;
+
+    SavedState( Parcelable superState ) {
+      super( superState );
+    }
+
+    private SavedState( Parcel in ) {
+      super( in );
+      scale = in.readFloat();
+      savedCenterX = in.readInt();
+      savedCenterY = in.readInt();
+    }
+
+    @Override
+    public void writeToParcel( Parcel out, int flags ) {
+      super.writeToParcel( out, flags );
+      out.writeFloat( scale );
+      out.writeInt( savedCenterX );
+      out.writeInt( savedCenterY );
+    }
+
+    public static final Parcelable.Creator<SavedState> CREATOR
+            = new Parcelable.Creator<SavedState>() {
+      public SavedState createFromParcel( Parcel in ) {
+        return new SavedState( in );
+      }
+
+      public SavedState[] newArray( int size ) {
+        return new SavedState[ size ];
+      }
+    };
+  }
+
+  /**
+   * The default {@code super.onSaveInstanceState} and {@code onRestoreInstanceState} don't
+   * restore the position on the map as expected (if the instance of {@link TileView} remains the
+   * same). For this reason and if a new {@link TileView} instance is created, we have to save
+   * the current scale and position on the map, to restore them later when the {@link TileView} is
+   * recreated.
+   */
+  @Override
+  public Parcelable onSaveInstanceState() {
+    Parcelable superState = super.onSaveInstanceState();
+    SavedState ss = new SavedState( superState );
+    ss.scale = getScale();
+    ss.savedCenterX = getScrollX() + getHalfWidth();
```

Comment:
In fact i think we have to save the center because on a config change, the top-left corner's position in pixel changes.